SoulMatchServer-main/src/main/java/com/soulmatch/ai/AIController.java

```diff
@@ -0,0 +1,90 @@
+package com.soulmatch.ai;
+
+import com.google.gson.*;
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AIController {
+
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+    private static final String API_ENDPOINT = "https://ai-danger-hot-or-not.hf.space/api/predict";
+    private static final HttpClient CLIENT = HttpClient.newBuilder().version(HttpClient.Version.HTTP_2).build();
+
+    public static int predictScore(String image) {
+        String prefix = "png";
+        if (image.toLowerCase().endsWith(".jpg") || image.toLowerCase().endsWith(".jpeg")) {
+            prefix = "jpeg";
+        }
+
+
+        AIData aiData = new AIData();
+
+        Path filesPath = Paths.get("files").resolve(image);
+        File file = new File(filesPath.toUri());
+
+        try (FileInputStream fileInputStreamReader = new FileInputStream(file)) {
```

Comment:
Make sure to do smaller commits instead of larger ones so it's easier to reverse if something goes wrong